### PR TITLE
FIX: remove hard-coded libstdc++

### PR DESCRIPTION
**Pull request by reneme**
In our setup we use a non-standard clang installation on macOS which comes with `libc++`. Hence, my default conan profile looks like that:

```
[...]
[settings]
os=Macos
[...]
compiler=clang
compiler.version=6.0
compiler.libcxx=libc++
[...]
```

Hence, all my conan dependencies are built against `libc++` (rather than `libstdc++`). Since this package hard-coded `libstdc++` for `clang` it didn't build for me.

When looking into the history of the adapted code, I found that `apple-clang` was actually removed from this hardcoded list before (presumably for a similar reason?). Unfortunately my non-standard compiler is classified as `clang` rather than `apple-clang`.

Bottom line: I'd argue that hard-coding the STL implementation to be used when building a conan dependency is never a good idea.

### CODE REVIEW

**Comment by solvingj** [comment]
you are absolutely correct. this was done long ago and we no longer do this in recipes.  the modern approach is to raise exception when required things are not set.  in this case, i'm not even sure that's what we should be doing. if we do that, it will likely involve a more specific conditional. ping @SSE4 @uilianries for feedback

**Comment by SSE4** [comment]
for me looks correct.
P.S. something weird with appveyor CI and ICU, I always see the same cancelled/failed build, is same happens for everyone or just for me? restart doesn't change anything.

**Comment by SSE4** [comment]
update: I've created appveyor support ticket
https://help.appveyor.com/discussions/problems/17185-build-seems-to-be-cancelling-itself

**Comment by SSE4** [comment]
this is blocked by too long vcvars's produced PATH, waiting for conan 1.8 which has a fix
